java/src/main/native/src/row_conversion.cu

```diff
@@ -63,6 +64,9 @@ constexpr auto NUM_TILES_PER_KERNEL_LOADED = 2;
 constexpr auto NUM_VALIDITY_TILES_PER_KERNEL = 8;
 constexpr auto NUM_VALIDITY_TILES_PER_KERNEL_LOADED = 2;
 
+constexpr auto NUM_STRING_ROWS_PER_BLOCK = 16;
+constexpr auto MAX_STRING_BLOCKS = 2147483647; // way to query CUDA?
```

Comment:
How about `std::numeric_limits<cudf::size_type>::max();`

Reply:
I thought about that and also doing the math listed in the cuda docs, but both seem easy to be wrong. The real thing we want to know is "how many blocks am I allowed to create" not how big can an int be.

Reply:
I'm not sure which value should be to be safe, but I think if this value depends on some other values then it should be computed using a (`constexpr`) function.

Reply:
I don't know if we have access to those values though. This is the number of blocks allowed in cuda. I don't know a way to get to that at runtime. It was originally 65k and was bumped for the x axis.